tests/net/dhcpv4/server/prj.conf

```diff
@@ -0,0 +1,16 @@
+CONFIG_NET_TEST=y
+CONFIG_ZTEST=y
+
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+
+CONFIG_NETWORKING=y
+CONFIG_NET_L2_DUMMY=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_IPV6=n
+CONFIG_NET_UDP=y
+CONFIG_NET_DHCPV4_SERVER=y
+
+# We need to set POSIX_API and use picolibc for eventfd to work
+CONFIG_POSIX_API=y
+CONFIG_PICOLIBC=y
```

Comment:
Isn't this the default?

Reply:
I've checked this, and libc defaults to picolib only if `CONFIG_POSIX_API` is enabled (otherwise extrernal libc is selected). So in theory we could leave POSIX_API only, but IMO I'd keep both to make the intent here clear.